### PR TITLE
Properly shutdown wl display

### PR DESCRIPTION
**Pull request by ikalco**
#### Describe your PR, what does it fix/add?
use wl_display_destroy when exiting
properly destroy wayland globals when doing above
destroy protocols before compositor cleanup
properly handle Hyprland specifics for shutting down

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
It works for me&trade;

#### Is it ready for merging, or does it need work?
yes

### CODE REVIEW

File: src/managers/eventLoop/EventLoopManager.cpp [hunk 1]
```diff
@@ -44,8 +50,6 @@ void CEventLoopManager::enterLoop() {
     }
 
     wl_display_run(m_sWayland.display);
-
-    Debug::log(LOG, "Kicked off the event loop! :(");
```
Comment (vaxerski) [inline src/managers/eventLoop/EventLoopManager.cpp:48]: why log removed :(

File: src/main.cpp [hunk 2]
```diff
@@ -157,11 +157,7 @@ int main(int argc, char** argv) {
     // If all's good to go, start.
     g_pCompositor->startCompositor();
 
-    g_pCompositor->m_bIsShuttingDown = true;
-
-    // If we are here it means we got yote.
-    Debug::log(LOG, "Hyprland reached the end.");
```
Comment (khaneliman) [inline src/main.cpp:163]: should this be removed too?

File: src/managers/eventLoop/EventLoopManager.cpp (hunk 1 above)
Comment (ikalco) [reply src/managers/eventLoop/EventLoopManager.cpp]: :)

**Comment by khaneliman** [comment]
lmao that commit message

**Review by vaxerski — APPROVED** [review]
thanks!

**Pull request merged**